src/test/java/de/retest/recheck/cli/subcommands/DiffIT.java

```diff
@@ -61,4 +68,23 @@ public void diff_should_print_differences() throws Exception {
 
 		assertThat( systemOutRule.getLog() ).contains( expected );
 	}
+
+	@Test
+	public void checkFilterNames_should_check_the_filter_names() throws Exception {
+		final String filterName = "positioning.filter";
+		final Diff cut = new Diff();
+		final List<Filter> filters = new ArrayList<>();
+		final Optional<Filter> filter = SearchFilterFiles.searchFilterByName( filterName );
+		filters.add( RecheckIgnoreUtil.loadRecheckIgnore() );
+		if ( filter.isPresent() ) {
+			filters.add( filter.get() );
+		}
+		final CompoundFilter actualCompoundFilter = cut.checkFilterNames();
+
+		for ( final Filter actualFilter : actualCompoundFilter.getFilters() ) {
+			for ( final Filter expectedFilter : filters ) {
+				assertThat( actualFilter ).isEqualToComparingFieldByFieldRecursively( expectedFilter );
```

Comment:
This could potentially break if you do not know which implementaton is behind `Filter`... Consider using its method, since it is not important that the filters are equal whereas that they provide the same output.